includes/class-newspack-listings-taxonomies.php

```diff
@@ -0,0 +1,451 @@
+					'public'        => true,
+					'rewrite'       => [ 'slug' => self::NEWSPACK_LISTINGS_TAXONOMIES['event'] ],
+					'show_in_menu'  => false, // Set to 'true' to show in WP admin for debugging purposes.
+					'show_in_rest'  => true,
+					'show_tagcloud' => false,
+					'show_ui'       => true,
+					'labels'        => get_post_type_object( Core::NEWSPACK_LISTINGS_POST_TYPES['generic'] )->labels,
+				],
+			],
+			'marketplace' => [
+				'post_types' => [
+					'post',
+					'page',
+					Core::NEWSPACK_LISTINGS_POST_TYPES['generic'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['marketplace'],
+				],
+				'config'     => [
+					'hierarchical'  => true,
+					'public'        => true,
+					'rewrite'       => [ 'slug' => self::NEWSPACK_LISTINGS_TAXONOMIES['marketplace'] ],
+					'show_in_menu'  => false, // Set to 'true' to show in WP admin for debugging purposes.
+					'show_in_rest'  => true,
+					'show_tagcloud' => false,
+					'show_ui'       => true,
+					'labels'        => get_post_type_object( Core::NEWSPACK_LISTINGS_POST_TYPES['marketplace'] )->labels,
+				],
+			],
+			'place'       => [
+				'post_types' => [
+					'post',
+					'page',
+					Core::NEWSPACK_LISTINGS_POST_TYPES['event'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['generic'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['marketplace'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['place'],
+				],
+				'config'     => [
+					'hierarchical'  => true,
+					'public'        => true,
+					'rewrite'       => [ 'slug' => self::NEWSPACK_LISTINGS_TAXONOMIES['place'] ],
+					'show_in_menu'  => false, // Set to 'true' to show in WP admin for debugging purposes.
+					'show_in_rest'  => true,
+					'show_tagcloud' => false,
+					'show_ui'       => true,
+					'labels'        => get_post_type_object( Core::NEWSPACK_LISTINGS_POST_TYPES['place'] )->labels,
+				],
+			],
+		];
+
+		// Register shadow taxonomies for each source post type.
+		foreach ( $shadow_taxonomies as $post_type_to_shadow => $shadow_taxonomy ) {
+			register_taxonomy(
+				self::NEWSPACK_LISTINGS_TAXONOMIES[ $post_type_to_shadow ],
+				$shadow_taxonomy['post_types'],
+				$shadow_taxonomy['config']
+			);
+		}
+	}
+
+	/**
+	 * Using the array of shadow taxonomies, get an array of corresponding post types being shadowed.
+	 *
+	 * @return array Array of post types being shadowed.
+	 */
+	public static function get_post_types_to_shadow() {
+		return array_values(
+			array_intersect_key(
+				Core::NEWSPACK_LISTINGS_POST_TYPES,
+				self::NEWSPACK_LISTINGS_TAXONOMIES
+			)
+		);
+	}
+
+	/**
+	 * Create shadow relationships between taxonomies and their posts.
+	 */
+	public static function create_shadow_relationship() {
+		add_action( 'wp_insert_post', [ __CLASS__, 'update_or_delete_shadow_term' ], 10, 2 );
+		add_action( 'before_delete_post', [ __CLASS__, 'delete_shadow_term' ] );
+	}
+
+	/**
+	 * When a Place changes status, add/update the shadow term if the status is `publish`, otherwise delete it.
```

Comment:
Why is only one of the four listing types handled here? After reverting a generic listing to draft, it was still available in the taxonomy sidebar on a post.

Reply:
This was an oversight (since I started off with just one post type for testing). Should be fixed in a6f0195.

Reply:
Not sure if that worked, I've found this issue:
1. Create a Place
2. Assign it to a Marketplace
3. Rename the place
4. Edit the Marketplace - there is now an additional Place taxonomy item - both old and new name are listed are assignable taxonomy options

Reply:
Also, the comment still mentions "Place" ;)